src/main/java/com/team2502/robot2020/subsystem/ShooterSubsystem.java

```diff
@@ -0,0 +1,80 @@
+package com.team2502.robot2020.subsystem;
+
+import com.revrobotics.*;
+import com.team2502.robot2020.Constants;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class ShooterSubsystem extends SubsystemBase {
+
+    private final CANSparkMax shooterRight;
+    private final CANSparkMax shooterLeft;
+
+    public final CANPIDController rightPID;
+    private final CANEncoder rightEncoder;
+
+    private boolean continuousAdjustmentToggled;
+
+    public ShooterSubsystem() {
+        shooterLeft = new CANSparkMax(Constants.RobotMap.Motors.SHOOTER_LEFT, CANSparkMaxLowLevel.MotorType.kBrushless);
+        shooterLeft.setSmartCurrentLimit(35);
+        shooterRight = new CANSparkMax(Constants.RobotMap.Motors.SHOOTER_RIGHT, CANSparkMaxLowLevel.MotorType.kBrushless);
+        shooterRight.setSmartCurrentLimit(35);
+        shooterLeft.follow(shooterRight, true);
+
+        rightPID = shooterRight.getPIDController();
+        rightEncoder = shooterRight.getEncoder();
+
+        continuousAdjustmentToggled = false;
+
+        setupPID();
+    }
+
+    @Override
+    public void periodic() { SmartDashboard.putNumber("Shooter Velocity", rightEncoder.getVelocity()); }
+
+    public void setShooterSpeedRPM(double speed) {
+        rightPID.setReference(speed, ControlType.kVelocity);
+        SmartDashboard.putNumber("Shooter Target Velocity", speed);
+    }
+
+    public void setShooterSpeedVision(double defaultSpeed, VisionSubsystem vision) {
+        if(vision.getArea() > 0) {
+            setShooterSpeedRPM(vision.getOptimalShooterSpeed());
+        }
+        else {
+            setShooterSpeedRPM(defaultSpeed);
+        }
+    }
+
+    public void setShooterSpeedVisionToggle(double defaultSpeed, VisionSubsystem vision){
+        while(continuousAdjustmentToggled){
```

Comment:
this while loop will block the whole method and stop the `ToggleShooterCommand` from finishing. because the scheduler "interlaces" commands (iirc), this will probably block the whole robot